script/one_time_scripts/remove_invalid_authorized_voters.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+require File.expand_path('../../config/environment', __dir__)
+
+VoteAccount.joins(:account_authority_histories)
+           .includes(:account_authority_histories)
+           .find_each do |va|
+  invalid_voters = va.account_authority_histories.select do |history|
+    if history.authorized_withdrawer_after == history.authorized_withdrawer_before
+      before_values = Array(history.authorized_voters_before&.values)
+      after_values = Array(history.authorized_voters_after&.values)
+
+      before_values.size == (before_values & after_values).size
```

Comment:
This is incorrect.
You're comparing only to before_values, so It will only work in the scenario:
before_values = ["test"]
after_values = ["test"] or after_values = ["test2"]  (equal number of elements before and after)

or
before_values = ["test", "test2"]
after_values = ["test"]
(more elements before than after)

but will not work in the following scenario:
before_values = ["test"]
after_values = ["test", "test2"]
(more elements after than before)
because the common part of the collection ("test") is the same as before_values. And before and after values are not at all the same, so it incorrectly detects this record to be removed. 

Instead, you should just sort and compare arrays.

<img width="537" alt="Zrzut ekranu 2023-06-13 o 21 46 08" src="https://github.com/brianlong/www.validators.app/assets/32059143/0b83b9f8-3383-40c9-9b76-acc8aa730cb0">


And please, write some tests for your code. This kind of issues should be detected by tests.

Reply:
Good catch. Agree, sorting and comparing way will do the thing, but I see we can remove one array from the another and see if any elements has left. This seems slightly faster.

Reply:
Good catch. Agree, sorting and comparing way will do the thing, but I see we can use `<=>` operator as well in order to compare arrays, which seems slightly faster.

Reply:
Do not substract arrays (I'm referring to your previous comment) - this solutions is prone to errrors, because it matters  which array is larger.
Comparing with spaceship operator will work, but remember it returns -1/0/1, instead of true/false. Also, you won't avoid sorting.

Reply:
I cannot avoid sorting. That's a pitty, but it's true.